app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -880,7 +874,7 @@ private void startNextFetch() {
     // extremely long method
 
     private void readyToProceed(final CommCareSession session) {
```

Comment:
It seems bad that we're receiving a session arg, but also creating our own session reference internally from the static pointer. 

Would prefer to pass an AndroidSessionWrapper into each method instead, and use that reference rather thn create a new one.